velox/connectors/hive/storage_adapters/s3fs/tests/S3InsertTest.cpp

```diff
@@ -0,0 +1,179 @@
+
+using namespace facebook::velox;
+using namespace facebook::velox::core;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::connector;
+using namespace facebook::velox::connector::hive;
+using namespace facebook::velox::dwio::common;
+using namespace facebook::velox::test;
+using namespace facebook::velox::filesystems;
+
+class S3InsertTest : public testing::Test, public VectorTestBase {
+ public:
+  static constexpr char const* kMinioConnectionString{"127.0.0.1:7000"};
+  static void SetUpTestSuite() {
+    minioServer_ = std::make_shared<MinioServer>(kMinioConnectionString);
+    minioServer_->start();
+
+    ioExecutor_ = std::make_unique<folly::IOThreadPoolExecutor>(3);
+    filesystems::registerS3FileSystem();
+    auto hiveConnector =
+        connector::getConnectorFactory(
+            connector::hive::HiveConnectorFactory::kHiveConnectorName)
+            ->newConnector(
+                kHiveConnectorId,
+                minioServer_->hiveConfig(),
+                ioExecutor_.get());
+    connector::registerConnector(hiveConnector);
+  }
+
+  static void TearDownTestSuite() {
+    filesystems::finalizeS3FileSystem();
+    unregisterConnector(kHiveConnectorId);
+    minioServer_->stop();
+    minioServer_ = nullptr;
+  }
+
+  static std::shared_ptr<MinioServer> minioServer_;
+  static std::unique_ptr<folly::IOThreadPoolExecutor> ioExecutor_;
+};
+std::shared_ptr<MinioServer> S3InsertTest::minioServer_ = nullptr;
+std::unique_ptr<folly::IOThreadPoolExecutor> S3InsertTest::ioExecutor_ =
+    nullptr;
+
+PlanNodePtr createInsertPlan(
+    PlanBuilder& inputPlan,
+    const RowTypePtr& outputRowType,
+    const std::string_view& outputDirectoryPath,
+    const std::vector<std::string>& partitionedBy = {},
+    const std::shared_ptr<HiveBucketProperty>& bucketProperty = {},
+    const connector::hive::LocationHandle::TableType& outputTableType =
+        connector::hive::LocationHandle::TableType::kNew,
+    const CommitStrategy& outputCommitStrategy = CommitStrategy::kNoCommit) {
+  auto insertTableHandle = std::make_shared<core::InsertTableHandle>(
+      kHiveConnectorId,
+      HiveConnectorTestBase::makeHiveInsertTableHandle(
+          outputRowType->names(),
+          outputRowType->children(),
+          partitionedBy,
+          bucketProperty,
+          HiveConnectorTestBase::makeLocationHandle(
+              outputDirectoryPath.data(), std::nullopt, outputTableType),
+          FileFormat::PARQUET));
+
+  auto insertPlan = inputPlan.tableWrite(
+      inputPlan.planNode()->outputType(),
+      outputRowType->names(),
+      nullptr,
+      insertTableHandle,
+      bucketProperty != nullptr,
+      outputCommitStrategy);
+  return insertPlan.planNode();
+}
+
+TEST_F(S3InsertTest, s3InsertTest) {
+  const int64_t kExpectedRows = 1'000;
+  const std::string_view kOutputDirectory{"s3://writedata/"};
+
+  auto rowType = ROW(
```

Comment:
would passing more datatypes (bool, string, complex types etc.) to the row help strengthen this test further?

Reply:
I don't think so. Same answer as using a fuzzer. S3FileSink is simply a wrapper around S3WriteFile.